### DISCUSSION TITLE
Input prompts from the file

### DISCUSSION THREAD

**Discussion by aniketgore** [comfyanonymous/ComfyUI#896] [Q&A]
This might be a feature already, but I couldn't find it.

I want to have a node that will iterate through a text file and feed one prompt as an input -> generate an image -> pickes up next prompt and do this until the prompts in the file are finished. A similar function in auto is prompt from file/textbox script.

**Comment by aniketgore** [comment]
Is this already existing?

**Comment by zhhezhhe** [comment]
same question

**Comment by MojoJojo43** [comment]
same question

**Comment by bluesky61** [comment]
I need one, too.

**Comment by ltdrdata** [comment]
> This might be a feature already, but I couldn't find it.
> 
> I want to have a node that will iterate through a text file and feed one prompt as an input -> generate an image -> pickes up next prompt and do this until the prompts in the file are finished. A similar function in auto is prompt from file/textbox script.

Are you referring to iterating through multiple prompt files in a directory and reading them, or are you referring to a situation where multiple prompts are contained within a single prompt file?

**Comment by ltdrdata** [comment]
Anyway.. I'd implemented this in `ComfyUI-Inspire-Pack`.

You can specify like this in `ComfyUI-Inspire-Pack/prompts/some_prompt_dir/some_prompt_name.txt`
```
positive:1girl is walking through street,
raincoat, yellow umbrella

negative:text, watermark
```

https://github.com/ltdrdata/ComfyUI-Inspire-Pack

![prompt-file](https://github.com/comfyanonymous/ComfyUI/assets/128333288/cf22c988-c936-42e4-9c52-978d53c96965)

**Reply by Centurion-Rome** [reply]
it would be great to have multiple prompts that were be looped so you could generate some template and run it overnight or maybe test results from other models

**Reply by tusharbhutt** [reply]
Yes, the WAS suite does this with the "Text Load Line From file" node:

https://github.com/WASasquatch/was-node-suite-comfyui#current-nodes

As usual, no real instructions, so place the full path **and file name** in slot one, connect it to an input and it will loop through the file sequentially.

Here's what it looks like when populated:

![image](https://github.com/comfyanonymous/ComfyUI/assets/26264462/309cd280-bca0-4c91-9f32-79b8f8923058)

**Comment by tusharbhutt** [comment]
Here's a four way prompt input:

1. Using OneButtonPrompt
2. User Input
3. Sequential Line from File
4. Random Line from File

![image](https://github.com/comfyanonymous/ComfyUI/assets/26264462/522811a2-d02c-4e02-8525-ccf3deb25a3d)


To get really creative, you can randomize the input to come from either OBP or a random line:

![image](https://github.com/comfyanonymous/ComfyUI/assets/26264462/b9e01ce7-798a-4fee-8f50-11ee2d6a163b)

**Comment by arcreigh** [comment]
ComfyUI Essentials has the Text Encode For Sampler Params node where you have a single text box and each prompt is separated by ---
however it only seems to function with their iteration of a sampler parameters block. Unfortunately it doesn't seem to be able to pass these prompts as raw text to other core nodes.

**Comment by ltdrdata** [comment]
This functionality is already covered by custom nodes, so it will be closed.
